### PR TITLE
Move wsgiref.types to _typeshed.wsgi

### PR DESCRIPTION
**Pull request by srittau**
Re-export the types from wsgiref.types for now to avoid breaking
existing code. wsgiref.types should be removed eventually.

Also, reduce the boilerplate description in _typeshed/wsgi.pyi as it
mirrors the description in _typeshed/__init__.pyi.

### CODE REVIEW

File: tests/stubtest_whitelists/py3_common.txt [hunk 1]
```diff
@@ -5,7 +5,8 @@ _importlib_modulespec
 _operator.methodcaller
 _threading_local.local.__new__
 _types
-_typeshed
+_typeshed    # typeshed-only module
+_typeshed.*
```
Comment (hauntsaninja) [inline tests/stubtest_whitelists/py3_common.txt:9]:
```suggestion
_typeshed.*  # Utility types for typeshed, doesn't exist at runtime
```

File: stdlib/2and3/wsgiref/types.pyi [hunk 2]
```diff
@@ -1,44 +1,3 @@
...
-#
-#   from typing import TYPE_CHECKING
-#
-#   if TYPE_CHECKING:
-#       from wsgiref.types import WSGIApplication
-#
-# This import is now only taken into account by the type checker. Consequently,
-# you need to use 'WSGIApplication' and not simply WSGIApplication when type
-# hinting your code.  Otherwise Python will raise NameErrors.
+# Obsolete, use _typeshed.wsgi directly.
```
Comment (JelleZijlstra) [inline stdlib/2and3/wsgiref/types.pyi:1]: Why do we need to keep this file around at all? Are there usages outside typeshed?

**Comment by hauntsaninja** [comment]
Oh oops, my page was stale, so I merged without seeing Jelle's comment.

**Comment by JelleZijlstra** [comment]
No harm, we can remove the file in another PR if we don't need it.

**Comment by hauntsaninja** [comment]
It looks like it is used outside of typeshed:
- some book: https://github.com/PacktPublishing/Mastering-Object-Oriented-Python-Second-Edition/blob/master/Chapter_13/ch13_e1_ex2.py#L19
- tornado: https://github.com/tornadoweb/tornado/blob/b0f2479eadb23de5bed90f5f1cad0f159fa3602f/tornado/wsgi.py#L44
- reddit: https://github.com/reddit/baseplate.py/blob/master/baseplate/server/queue_consumer.py#L31

**Comment by spladug** [comment]
Just stumbled upon this. As the author of the reddit reference, I'd be happy to cut that tie if that's helpful here. Is it recommended just to define those types ourselves? Import from the private-looking `_typeshed` module? Some even better third option?

**Comment by srittau** [comment]
@spladug It's fine to import from `_typeshed`. And while we don't give compatibility guarantees, I don't expect much changes, once we have introduces a type. I hope that some of these types make it to the standard library eventually.

**Comment by spladug** [comment]
Got it. Will do. Thank you!

**Pull request merged**